source/dub/recipe/io.d

```diff
@@ -72,7 +73,27 @@ PackageRecipe parsePackageRecipe(string contents, string filename, string parent
 
 	ret.name = default_package_name;
 
-	if (filename.endsWith(".json")) parseJson(ret, parseJsonString(contents, filename), parent_name);
+	if (filename.endsWith(".json"))
+	{
+		ret = parseConfigString!PackageRecipe(contents, filename, StrictMode.Warn);
```

Comment:
what's StrictMode.warn? would it only log to the console and not throw any exception like it did before?

Execution should abort on invalid configs.

Reply:
Strict mode *only* checks for keys found in the YAML document that are not found in the configuration.
Currently dub flatly ignore those (on purpose). As I mentioned in the OP, that's terrible.
When I turned on strict mode, I found one error on my package cache (which was almost empty) https://github.com/symmetryinvestments/dud/pull/46
If this becomes the default for `dub`, there needs to be a way to warn users before it turns into an error. All the other errors (format, syntax, logical) are still errors and treated as such, strict mode only applies to keys the configuration is not expecting.